### PR TITLE
FE React for Rails

### PR DESCRIPTION
**Pull request by Axedez**
[Trello Task](https://trello.com/_____)

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [x] screenshot of feature is added if task has UI
image

### CODE REVIEW

File: spec/requests/react_test_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe "ReactTests", type: :request do
+  describe "GET /index" do
+    it "returns http success" do
+      get "/react-test/index"
+      expect(response).to have_http_status(:success)
+    end
+  end
+
```
Comment (sviatoslav-krupa) [inline spec/requests/react_test_request_spec.rb:9]: remove that empty line

File: app/javascript/packs/reactrailsujs.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Support component names relative to this directory:
+var componentRequireContext = require.context("components", true);
+var ReactRailsUJS = require("react_ujs");
+ReactRailsUJS.useContext(componentRequireContext);
```
Comment (sviatoslav-krupa) [inline app/javascript/packs/reactrailsujs.js:4]: add new line

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    return (
+      <div>
+        <h1>Test React App</h1>
+        <p>Hello world!</p>
+      </div>
+    );
+  }
+}
+
+export default App
```
Comment (sviatoslav-krupa) [inline app/javascript/components/App.js:14]:
```suggestion
import React, {Component} from 'react';

class App extends Component {
  render () {
    return (
      <div>
        <h1>Test React App</h1>
        <p>Hello world!</p>
      </div>
    );
  }
}

export default App;
```
Always put `;` at the end of each line.

**Review by sviatoslav-krupa — CHANGES_REQUESTED** [review]

File: Gemfile [hunk 4]
```diff
@@ -50,6 +50,7 @@ gem 'kaminari'
 gem 'premailer-rails'
 gem 'pundit'
 gem 'rails-i18n'
+gem 'react-rails'
```
Comment (marikrebega) [inline Gemfile:53]: please add gem version

**Review by Andrii-Kozak — APPROVED** [review]

**Review by mmarusyk — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**